src/porepy/viz/exporter.py

```diff
@@ -290,191 +409,563 @@ def write_pvd(
+        2. Unify the data format. Store data in dictionaries, with keys given by
+            grids/edges and names, and values given by the data arrays.
+
+        Parameters:
+            data (Union[DataInput, List[DataInput]], optional): data
+                provided by the user in the form of strings and/or tuples
+                of grids/edges.
+
+        Returns:
+            Tuple[SubdomainData, InterfaceData]: Subdomain and edge data decomposed and
+                brought into unified format.
         """
-        # No need of special naming, create the folder
-        name = self._make_folder(self.folder_name, self.file_name)
-        name = self._make_file_name(name, time_step)
 
-        # Provide an empty dict if data is None
+        # Convert data to list, while keeping the data structures provided,
+        # or provide an empty list if no data provided
         if data is None:
-            data = dict()
+            data = list()
+        elif not isinstance(data, list):
+            data = [data]
 
-        fields = Fields()
-        if len(data) > 0:
-            fields.extend([Field(n, v) for n, v in data.items()])
+        # Initialize container for data associated to nodes
+        subdomain_data: SubdomainData = dict()
+        interface_data: InterfaceData = dict()
 
-        grid_dim = self.grid.dim * np.ones(self.grid.num_cells, dtype=int)
+        # Auxiliary function transforming scalar ranged values
+        # to vector ranged values when suitable.
+        def toVectorFormat(value: np.ndarray, g: pp.Grid) -> np.ndarray:
+            """
+            Check whether the value array has the right dimension corresponding
+            to the grid size. If possible, translate the value to a vectorial
+            object, But do nothing if the data naturally can be interpreted as
+            scalar data.
+            """
+            # Make some checks
+            if not value.size % g.num_cells == 0:
+                raise ValueError("The data array is not compatible with the grid.")
 
-        fields.extend([Field("grid_dim", grid_dim)])
+            # Convert to vectorial data if more data provided than grid cells available,
+            # and the value array is not already in vectorial format
+            if not value.size == g.num_cells and not (
+                len(value.shape) > 1 and value.shape[1] == g.num_cells
+            ):
+                value = np.reshape(value, (-1, g.num_cells), "F")
 
-        self._write(fields, name, self.meshio_geom)
+            return value
 
-    def _export_gb(self, data: List[str], time_step: float) -> None:
-        """Export the entire GridBucket and additional data to vtu.
+        # Auxiliary functions for type checking
 
-        Parameters:
-            data (List[str]): Data to be exported in addition to default GridBucket data.
-            time_step (float) : Time step, to be appended at the vtu output file.
+        def isinstance_interface(e: Interface) -> bool:
```

Comment:
I do it here - my understanding is that you were asking what I have been wanting to achieve and how I did it:

My main goal was to perform the python routine isinstance on user-defined types, which mainly are of tuple type (e.g., `Interface`). However, isinstance, does not work on types defined using typing. It can only recognize objects that have been created using an constructor of an object. When however defining inline a tuple `a=("tmp", 2)` and the type `customTuple=Tuple[str,int]`, checking for `isinstance(a, customTuple)` is not possible.

A rather short solution to this is asking for the type of each argument of `a`, using `list(map(type, a))` which returns `[str,int]` in this case. 

When checking whether an argument is a list of grids, an additional check has be performed, as a list of grids will in the first place simply be identified as `list`. Thus, each element of this list is checked as well.